javaparser-core/src/main/java/com/github/javaparser/JavaParser.java

```diff
@@ -91,7 +89,7 @@ private GeneratedJavaParser getParserForProvider(Provider provider) {
         astParser.setTabSize(configuration.getTabSize());
         astParser.setStoreTokens(configuration.isStoreTokens());
         if (configuration.getLanguageLevel() != null) {
-            switch (configuration.getLanguageLevel()) {
+            switch(configuration.getLanguageLevel()) {
```

Comment:
Ah, I can see why you're building in that dirty detection. These files shouldn't be touched! The scope for the generators is: the AST nodes and the visitors. If you would like to extend that, please discuss a bit first so I'm not completely surprised :-D